ql/src/test/queries/clientpositive/union_flatten_subdir.q

```diff
@@ -0,0 +1,19 @@
+set hive.tez.union.flatten.subdirectories=true;
+
+create external table union_src1 (value string) partitioned by (key int);
+create external table union_src2 (value string) partitioned by (key int);
+create external table union_target (value string) partitioned by (key int);
+
+insert into table union_src1 partition (key = 1) values ("val1");
+insert into table union_src1 partition (key = 2) values ("val2");
+
+insert into table union_src2 partition (key = 1) values ("val3");
+insert into table union_src2 partition (key = 3) values ("val4");
+
+insert into table union_target
```

Comment:
create different tables for different scenarios:

union_target_acid_directinsert_flattened
union_target_mm_directinsert_flattened
union_target_nonacid_directinsert_flattened
union_target_acid_nodirectinsert_flattened
union_target_mm_nodirectinsert_flattened
union_target_nonacid_nodirectinsert_flattened

union_target_acid_directinsert_noflattened
union_target_mm_directinsert_noflattened
union_target_nonacid_directinsert_noflattened
union_target_acid_nodirectinsert_noflattened
union_target_mm_nodirectinsert_noflattened
union_target_nonacid_nodirectinsert_noflattened